tools/generator/ClassGen.cs

```diff
@@ -59,6 +59,7 @@ public ManagedClassGen (TypeDefinition t)
 		
 		public override string BaseType {
 			get { return nominal_base_type != null ? nominal_base_type.FullNameCorrected () : null; }
+			set { throw new NotImplementedException (); }
```

Comment:
It is not required (no need to do unless you are making further changes) but a suggested change is to use NotSupportedException instead of NIE. NIE means "we plan to implement" or "any derived classes are supposed to implement".